invenio_app_ils/ui/src/invenio_app_ils/forms/core/BaseForm/BaseForm.js

```diff
@@ -67,14 +105,24 @@ export class BaseForm extends Component {
         )}
         <Formik
           initialValues={this.initialValues}
-          onSubmit={this.onSubmit}
+          onSubmit={this.props.onSubmit || this.onSubmit}
           validationSchema={this.validationSchema}
-          render={({ isSubmitting, handleSubmit }) => (
-            <Form onSubmit={handleSubmit} loading={isSubmitting}>
+          render={({ isSubmitting, handleSubmit, submitForm, values }) => (
+            <Form
+              onSubmit={event => this.submitForm(event, 'submit', submitForm, values)}
+              loading={isSubmitting}
+            >
+              <ErrorMessage />
               {this.props.children}
-              <Button color="green" disabled={isSubmitting} type="submit">
-                Submit
-              </Button>
+              {buttons ? this.renderButtons(isSubmitting, submitForm, values) : (
+                <Button
+                  color="green"
```

Comment:
please do not hardcode colors on buttons or anything, it will be done with theming. You can set primary prop maybe (it won't be orange anymore)